src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+        try
+        {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(fastqName));
+            while ( readItr.hasNext() )
+            {
+                GATKRead read = readItr.next();
+                writer.write("@");
+                writer.write(read.getName());
+                writer.write(read.isFirstOfPair()?":1":":2");
+                writer.newLine();
+                writer.write(read.getBasesString());
+                writer.newLine();
+                writer.write('+');
+                writer.newLine();
+                for ( byte b : read.getBaseQualities() )
+                    writer.write(33+b);
+                writer.newLine();
+            }
+            writer.close();
+        }
+        catch ( IOException e )
+        {
+            throw new GATKException("Can't write "+fastqName+'.',e);
+        }
+    }
+
+    private static final class EventLocus implements Comparable<EventLocus>
+    {
+        private final int locusStart;
+        private final int locusEnd;
+        private long id = 0; // used as a uniquifier
+
+        public EventLocus( final int someLocusStart, final int intervalWidth )
+        {
+            locusStart = someLocusStart;
+            locusEnd = someLocusStart + intervalWidth;
+        }
+
+        public int getLocusStart() { return locusStart; }
+        public int getLocusEnd() { return locusEnd; }
+
+        public EventLocus makeUnique( long someId )
+        {
+            if ( id == 0 )
+                id = someId;
+            return this;
+        }
+
+        public int compareTo( final EventLocus ee )
+        {
+            int result = Integer.compare(locusStart,ee.locusStart);
+            if ( result == 0 ) result = Integer.compare(locusEnd,ee.locusEnd);
+            if ( result == 0 ) result = Long.compare(id,ee.id);
+            return result;
+        }
+    }
+
+    // Class to detect a split read:  that's a read where one part of the read maps well to reference, but
+    // another portion maps elsewhere or not at all.
+    // It returns the approximate locus of the breakpoint if it can find one.
+    // Class assumes that all reads presented to it are mapped.
+    private static final class SplitReadDetector
+    {
+        private static final int MIN_SOFT_CLIP_LEN = 30; // minimum length of an interesting soft clip
+        private static final int MIN_INDEL_LEN = 25; // minimum length of an interesting indel
+        private static final int SOFT_CLIP_LOCUS_WIDTH = 4; // uncertainty in event locus for soft clip
+        private static final byte MIN_QUALITY = 15; // minimum acceptable quality in a soft-clip window
+
+        public EventLocus getLocusIfReadIsSplit( GATKRead read )
+        {
+            final List<CigarElement> cigarElements = read.getCigar().getCigarElements();
+            final ListIterator<CigarElement> itr0 = cigarElements.listIterator();
+            if ( itr0.hasNext() )
```

Comment:
If someone were to write a ReverseListIterator that sat on top of a ListIterator to present a reversed view of the underlying List (probably about 30 lines of trivial boilerplate) I could eliminate a few lines of similar code.  Not worthwhile for me to do in isolation.